### ISSUE TITLE
Modularize wasm crate logic 

### ISSUE THREAD

**Issue by TalDerei** [penumbra-zone/penumbra#3431]
Structuring the `build_action` method within the `WasmPlanner` introduces a challenge when calling it from **web-workers**. It means that an instance of the `WasmPlanner` struct must be passed to the web-workers by either:

- **(1)** passing in the individual fields that comprise the `WasmPlanner` struct, and manually reconstructing an instance of the object within the web-worker, 
- **(2)** passing the complex object which internally invokes the [structured-clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm) to perform a deep-copy. 

These methods introduce unnecessary overhead and complexity. Instead, I propose relocating the `build_action` method to a dedicated [build](https://github.com/penumbra-zone/penumbra/blob/wasm-builder/crates/wasm/src/build.rs) file since its design doesn't depend on the `WasmPlanner`. This enables the `build_action` to be called directly from within the web-worker. 

As we expose more of the underlying proving system logic in wasm, the build file can be used to hold that logic.

**Comment by hdevalence** [comment]
I'm not sure I follow — why do these interact at all?

The Planner code produces the TransactionPlan and doesn't need to be parallel at all, it's not expensive to run. Once it finishes, the TransactionPlan has all the information needed to build the transaction, so the build methods should only need access to the TransactionPlan and the WitnessData, as described in the previous PRs. Where does the planner come in?

**Comment by hdevalence** [comment]
Oh, I see, is the point that the wasm code previously attached the building logic to the planner, even though planning and building are supposed to be separate phases?

**Comment by TalDerei** [comment]
> Oh, I see, is the point that the wasm code previously attached the building logic to the planner, even though planning and building are supposed to be separate phases?

Yeah exactly, the building logic shouldn't be under the planner. Separating them is cleaner and simplifies the `build_action` calls on the web-side.
